pkg/datadeliverystatus/datadeliverystatusfunctions.go

```diff
@@ -39,6 +45,7 @@ func (client *Client) patch(payload []byte, url string) (string, error) {
 	return string(body), nil
 }
 
+// These functions don't seem to deal with whitespaces and empty strings, not sure if this is intentional
```

Comment:
This is most likely out of scope for the ticket as it will involve adding new code to existing code. It might be good to add code to validate the string, "state" and make sure it matches up with our expected "state" values (see https://github.com/ONSdigital/blaise-data-delivery-status#states)